OpenRA.Platforms.Default/Shader.cs

```diff
@@ -18,24 +18,16 @@ namespace OpenRA.Platforms.Default
 {
 	sealed class Shader : ThreadAffine, IShader
 	{
-		public const int VertexPosAttributeIndex = 0;
-		public const int TexCoordAttributeIndex = 1;
-		public const int TexMetadataAttributeIndex = 2;
-		public const int TintAttributeIndex = 3;
-
 		readonly Dictionary<string, int> samplers = new();
 		readonly Dictionary<int, int> legacySizeUniforms = new();
 		readonly Dictionary<string, int> uniformCache = new();
 		readonly Dictionary<int, ITexture> textures = new();
 		readonly Queue<int> unbindTextures = new();
+		readonly IShaderBindings bindings;
 		readonly uint program;
 
-		static uint CompileShaderObject(int type, string name)
+		static uint CompileShaderObject(int type, string code, string name)
 		{
-			var ext = type == OpenGL.GL_VERTEX_SHADER ? "vert" : "frag";
-			var filename = Path.Combine(Platform.EngineDir, "glsl", name + "." + ext);
-			var code = File.ReadAllText(filename);
-
 			var version = OpenGL.Profile == GLProfile.Embedded ? "300 es" :
```

Comment:
Can this move into `ShaderBindings` to avoid magic manipulation of shader code that implementations can't control?

Reply:
I need to change the scope of this class to `OpenRA.Platforms.Default` or pass the profile

Reply:
it's actually hard to do either. There's a scope issues here

Reply:
This is currently dependency injected into a scope that has no idea about opengl